### PR TITLE
chore: Loosen fsspec requirements to allow recent releases

### PR DESCRIPTION
**Pull request by cburroughs**
(I'm not sure the project has a super consistent pattern for when to specify a maximum version, but was going for the smallest possible change.)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -155,7 +155,7 @@
         "grpcio-testing>=1.56.2,<2",
         "minio==7.1.0",
         "mock==2.0.0",
-        "moto",
+        "moto<5",
```
Comment (cburroughs) [inline setup.py:158]: 5 changed around the api and current tests do not pass without modification.

**Comment by sudohainguyen** [comment]
the last time we updated fsspec (from my commits), I got stuck when compiling by pyarrow deps.
But now we upgraded pyarrow successfully and It's fine to upgrade fsspec 🙌

File: setup.py (hunk 1 above)
Comment (tokoko) [reply setup.py:158]: I also stumbled into this, any idea why CI isn't failing as well?

Comment (cburroughs) [reply setup.py:158]: I *suspect* that some transitive dependency was previously holding it back.

Comment (sudohainguyen) [reply setup.py:158]: Moto v5 just released last week though, that's why CI succeeded

**Comment by cburroughs** [comment]
What does this PR need to move forward?

**Comment by sudohainguyen** [comment]
Just wanted to do integration tests since this issue is related to dependencies.

So closing this now? 🤔

**Comment by cburroughs** [comment]
cc @franciscojavierarceo per dev call

**Review by franciscojavierarceo — APPROVED** [review]
lgtm

**Pull request merged**